src/core/style/_elevation.scss

```diff
@@ -0,0 +1,186 @@
+);
+
+$_ambient-elevation-map: (
+  0: '0px 0px 0px 0px #{$_ambient-color}',
+  1: '0px 1px 3px 0px #{$_ambient-color}',
+  2: '0px 1px 5px 0px #{$_ambient-color}',
+  3: '0px 1px 8px 0px #{$_ambient-color}',
+  4: '0px 1px 10px 0px #{$_ambient-color}',
+  5: '0px 1px 14px 0px #{$_ambient-color}',
+  6: '0px 1px 18px 0px #{$_ambient-color}',
+  7: '0px 2px 16px 1px #{$_ambient-color}',
+  8: '0px 3px 14px 2px #{$_ambient-color}',
+  9: '0px 3px 16px 2px #{$_ambient-color}',
+  10: '0px 4px 18px 3px #{$_ambient-color}',
+  11: '0px 4px 20px 3px #{$_ambient-color}',
+  12: '0px 5px 22px 4px #{$_ambient-color}',
+  13: '0px 5px 24px 4px #{$_ambient-color}',
+  14: '0px 5px 26px 4px #{$_ambient-color}',
+  15: '0px 6px 28px 5px #{$_ambient-color}',
+  16: '0px 6px 30px 5px #{$_ambient-color}',
+  17: '0px 6px 32px 5px #{$_ambient-color}',
+  18: '0px 7px 34px 6px #{$_ambient-color}',
+  19: '0px 7px 36px 6px #{$_ambient-color}',
+  20: '0px 8px 38px 7px #{$_ambient-color}',
+  21: '0px 8px 40px 7px #{$_ambient-color}',
+  22: '0px 8px 42px 7px #{$_ambient-color}',
+  23: '0px 9px 44px 8px #{$_ambient-color}',
+  24: '0px 9px 46px 8px #{$_ambient-color}'
+);
+
+/**
+ * The css property used for elevation. In most cases this should not be changed. It is exposed
+ * as a variable for abstraction / easy use when needing to reference the property directly, for
+ * example in a `will-change` rule.
+ */
+$md-elevation-property: box-shadow !default;
+
+/**
+ * The default duration value for elevation transitions.
+ */
+$md-elevation-transition-duration: 280ms !default;
+
+/**
+ * The default easing value for elevation transitions.
+ */
+$md-elevation-transition-timing-function: $md-fast-out-slow-in-timing-function;
+
+/**
+ * Applies the correct css rules to an element to give it the elevation specified by $zValue.
+ * The $zValue must be between 0 and 24.
+ */
+@mixin md-elevation($zValue) {
+  @if type-of($zValue) != number or not unitless($zValue) {
+    @error "$zValue must be a unitless number";
+  }
+  @if $zValue < 0 or $zValue > 24 {
+    @error "$zValue must be between 0 and 24";
+  }
+
+  #{$md-elevation-property}: #{map-get($_umbra-elevation-map, $zValue)},
+                             #{map-get($_penumbra-elevation-map, $zValue)},
+                             #{map-get($_ambient-elevation-map, $zValue)};
+}
+
+/**
+ * Returns a string that can be used as the value for a `transition` property for elevation.
+ * Calling this function directly is useful in situations where a component needs to transition
+ * more than one property.
+ *
+ * ```scss
+ * .foo {
+ *   transition: md-elevation-transition-rule(), opacity 100ms ease;
+ *   will-change: $md-elevation-property, opacity;
+ * }
+ * ```
+ */
+@function md-elevation-transition-rule(
+    $duration: $md-elevation-transition-duration,
+    $easing: $md-elevation-transition-timing-function) {
+  @return #{$md-elevation-property} #{$duration} #{$easing};
+}
```

Comment:
It would be nice if we could think of a name that more clearly distinguishes this from the below `md-elevation-transition`...

`md-elevation-transition-definition`? `md-get-elevation-transition-content`?